### PR TITLE
Update licence headers

### PR DESCRIPTION
**Pull request by dmey**
This PR adds/updates the licence header and copyright notice. @tomgrylls and @ivosuter please can you check and let me know.

~There are however a few files which have no authorship information and are not included in the dales repo therefore I am not sure which copyright should apply (i.e. DALES or uDALES). Could you please check the following files and let me know?~

~1. These seem to be IBM related so I think they should be DALES but please double check:~

~modibm.f90~
~modibmdata.f90~
~modinlet.f90~
~modinletdata.f90~
~modsave.f90~

~2.  These I don't know:~

~wf_gr.f90~
~wf_uno.f90~
~wfmneutral.f90~

EDIT: point 1 is addressed in 51cff24 and point 2 in e580661

### CODE REVIEW

**Comment by tomgrylls** [comment]
@ivosuter is best placed to advise on the first set you have listed. From my understanding, these are modules that were added by either @ivosuter or Jasper Tomas. They are not all IBM related but have been added since this branched from DALES.

The second set are all wall functions that were added by @ivosuter and me so should be uDALES.

**Comment by ivosuter** [comment]
modibm.f90
modibmdata.f90
modinlet.f90
modinletdata.f90
modsave.f90

Are all from Jasper and probably modified by us later (originate from DALESURBAN_136, Jaspers version we built upon). None of them are in DALES 3.2 or DALES 4.0, so they must have been added by Jasper himself.

**Comment by dmey** [comment]
> Are all from Jasper and probably modified by us later (originate from DALESURBAN_136, Jaspers version we built upon). None of them are in DALES 3.2 or DALES 4.0, so they must have been added by Jasper himself.

OK thanks -- will use the DALES header for these then.

**Comment by dmey** [comment]
@ivosuter @tomgrylls this is now ready for review could approve/let me know for any changes please?

**Comment by tomgrylls** [comment]
This all looks okay to me. One question before approving. Modules like `modboundary` are now highly modified versions of the original DALES modules. Is this implied through the licensing text or should it be said that modules like that are edited versions of the originals too?

**Comment by dmey** [comment]
>One question before approving. Modules like modboundary are now highly modified versions of the original DALES modules. Is this implied through the licensing text or should it be said that modules like that are edited versions of the originals too?

So if these were originally provided in DALES and you modified them (no matter by how much) the original licence and copyright still applies. You can only change a licence and copyright if this is created by yourself.

File: src/modstartup.f90 [hunk 1]
```diff
@@ -13,6 +13,22 @@
 !!  \todo documentation
 !!  \par Revision list
 !
+!
```
Comment (tomgrylls) [inline src/modstartup.f90:16]: Missing "This file is part of DALES"

File: src/modsubgriddata.f90 [hunk 2]
```diff
@@ -11,6 +11,22 @@
 !!  \author Thijs Heus,MPI-M
 !!  \par Revision list
 !!  \todo Documentation
+!
```
Comment (tomgrylls) [inline src/modsubgriddata.f90:14]: Missing "This file is part of DALES"

File: src/scalsource.f90 [hunk 3]
```diff
@@ -2,7 +2,23 @@
 !!tg3315, 8 Apr 2016
```
Comment (tomgrylls) [inline src/scalsource.f90:2]: Author: Tom Grylls to be consistent with other files.

Other files I have authored (e.g. modchem) do not have the author name. For consistency perhaps add my name to that. Or also I am equally happy for my author names to be removed across all files as many files have had secondary input across the uDALES team, which isn't captured.

File: src/modstartup.f90 (hunk 1 above)
Comment (dmey) [reply src/modstartup.f90:17]: fixed by 29d4175

File: src/modsubgriddata.f90 (hunk 2 above)
Comment (dmey) [reply src/modsubgriddata.f90:14]: fixed by 29d4175

File: src/scalsource.f90 (hunk 3 above)
Comment (dmey) [reply src/scalsource.f90:2]: Happy either way -- these files/headers would require a lot of clean up. In this PR I was just trying to address the licencing and copyright issue but I can remove the username and just leave the uDALES Team bit if you are happy with that. What about the other authors/contributors? Shall we leave them as they are for now and address that in a future PR?

Comment (tomgrylls) [reply src/scalsource.f90:2]: Yes that is fine by me - let's remove the username here now for consistency. Authors/ contributors can be adjusted if needed in a future PR.

Comment (dmey) [reply src/scalsource.f90:2]: There are a lot of instances in a lot of files as comments -- do I need to remove those too or shall I just remove the ones in the header?

Comment (tomgrylls) [reply src/scalsource.f90:2]: No I just meant in this header

Comment (tomgrylls) [reply src/scalsource.f90:2]: Thanks!

Comment (dmey) [reply src/scalsource.f90]: 👍 Cool. Fixed in 4725cf6. BTW re generating paper add `from branch joss-paper` to the usual command.

Comment (tomgrylls) [reply src/scalsource.f90]: Thanks!!!

**Review by tomgrylls — APPROVED** [review]
Happy to approve this now.

**Pull request merged**